addons/website_sale_collect/controllers/main.py

```diff
@@ -27,8 +27,20 @@ def _prepare_product_values(self, product, category, search, **kwargs):
 
     def _prepare_checkout_page_values(self, order_sudo, **query_params):
         """ Override of `website_sale` to include the unavailable products for the selected pickup
-        location. """
+        location and set the pickup location when there is only one warehouse available. """
         res = super()._prepare_checkout_page_values(order_sudo, **query_params)
+        in_store_methods = [
+            dm for dm in order_sudo._get_delivery_methods() if dm.delivery_type == 'in_store'
+        ]
+
+        for delivery_method in in_store_methods:
```

Comment:
Even if I have multiple in_store delivery method we can only have on set on order so doing this loop seems not useful

Like here I found 3 in_store_methods then first it'll first delivery method warehouse then overwrite with second and then third so actually it is only setting 3rd DM warehouse on order
So IMO you can just get only DM method which is in_store and set that on the order
 I am not sure they want to set default location for all DM in_store :thinking: (Check with PO)

Also this means if on pick in_store method with one warehouse if publish then it'll set it buy default instead standard method and it there it two warehouse then it'll set standard method by default
I am not sure change default method to in_store method was part of task and is good idea

Reply:
> Also this means if on pick in_store method with one warehouse if publish then it'll set it buy default instead standard method and it there it two warehouse then it'll set standard method by default
> I am not sure change default method to in_store method was part of task and is good idea
> 

We need to set the pickup location for the order during the preprocessing phase. This is because, when setting the default location for the delivery method, the page requires a reload, which is not ideal from a UX perspective.

Furthermore, in order to assign the pickup location to the order, we must also set the delivery method to "Pick Up in Store".

Reply:
I am not sure it would be great idea I understand you POV from technical side and UX
Lets say I open Ecom site for clothes in Gandinagar and enable pick up in store for my customers
How many do you think outside Gandinagar would prefer to pick up in store. So they have to make sure they update delivery method every time they don't want to pick up in store before your changes it was depend on sequence so if I have shiprocket first DM then it'll select default shiprocket but after your changes it'll always select default pick up in store regardless of sequence

@Jean-boje @shbh-odoo can I have you opinion on this one also?